src/hooks/usePositions.tsx

```diff
@@ -161,6 +161,7 @@ export const usePositions = (options: OptionsToSearch) => {
             userBalanceERC1155Numbered: Number(userBalanceERC1155WithDecimals),
             userBalanceERC20Numbered: Number(userBalanceERC20WithDecimals),
             collateralTokenERC1155: collateralTokenFound.length && collateralTokenFound[0],
+            token: collateralTokenFound.length && collateralTokenFound[0],
             collateralTokenSymbol:
               collateralTokenFound.length && collateralTokenFound[0]
```

Comment:
There were some name collisions here, collateralTokenERC20 is related to the wrapped ERC20 and his balance and not the token information, so instead of using `useWithToken` hook outside, I leveraged the fact that usePositions also get token information for each position.